### DISCUSSION TITLE
How to configure SEQ to use seq.mydomain.com

### DISCUSSION THREAD

**Discussion by Verbient** [datalust/seq-tickets#1360] [Help]
I have tried all things to host seq to [seq.mydomain.com](url) but failed. It there any comprehensive tutorial on it?

**Comment by nblumhardt** [comment]
Hi @Verbient 👋 

https://docs.datalust.co/docs/urls is the best source of information if you're doing this on Windows. Note that you'll also need some kind of DNS to point `seq.mydomain.com` to your server, and will also have to ensure that the appropriate HTTP or HTTPS firewall port is open.

If you're hitting issues it would be great to get an idea of what you've tried, what tech stack you're using, and how the current setup is failing.

HTH!
Nick

**Comment by Verbient** [comment]
I am getting the following error:

C:\Users\Administrator>seq install --listen="http://localhost:5341" -u "admin" -p "PassWord"
Installing service...
Updating the configuration found in C:\ProgramData\Seq\Seq.json...
Ensuring admin is granted 'Log on as a Service' rights...
Could not install the service: No mapping between account names and security IDs was done. (0x80070534)

**Reply by nblumhardt** [reply]
Thanks for the update. Is `admin` an account on your local Windows machine? The `-u` and `-p` options are for service account logon, not the username and password for Seq itself. Let me know if this helps!

**Reply by Verbient** [reply]
Thanks for pointing out this fact. I have corrected myself. However still I am getting the following error when I try to point to the following URL which is a live URL. Do I have to make any changes in IIS. I can assure you that I am providing the correct username and password:

C:\Users\Administrator>seq service install --listen="https://seq.xxxxx.com" -u "Administrator" -p "xxxxxx"
Installing service...
Updating the configuration found in C:\ProgramData\Seq\Seq.json...
Ensuring Administrator is granted 'Log on as a Service' rights...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq\Logs...
Adding URL reservation at https://seq.xxxxx.com/ for Administrator...

Url reservation add failed, Error: 87
The parameter is incorrect.


Could not add URL reservation for https://seq.xxxxx.com/: `netsh` returned 1; ignoring
[SC] CreateService FAILED 1057:

The account name is invalid or does not exist, or the password is invalid for the account name specified.

Could not install the service: Service setup failed

**Reply by nblumhardt** [reply]
Thanks for your reply. I think you may need to use `machine\username` instead of just `username`, e.g. `-u "YOURMACHINE\Administrator"` - looks like we need to update the instructions 👍

**Reply by Verbient** [reply]
I tried using machine\username but got the following error this time:

C:\Windows\system32>seq service install --listen "http://seq.xxxx.com" -u "XXX/Administrator" -p "xxx"
Installing service...
Updating the configuration found in C:\ProgramData\Seq\Seq.json...
Ensuring XXX/Administrator is granted 'Log on as a Service' rights...
Could not install the service: No mapping between account names and security IDs was done. (0x80070534)

**Reply by nblumhardt** [reply]
Thanks for your reply; you'll need a backslash between the domain and user parts of the username (not a forwardslash). Hope this helps!

**Reply by Verbient** [reply]
This time the following error occured. I also used the correct port number:

C:\Windows\system32>seq service install --listen="https://seq.xxx.com:port" -u "xxx\Administrator" -p "xxx"
Installing service...
Updating the configuration found in C:\ProgramData\Seq\Seq.json...
Ensuring xxx\Administrator is granted 'Log on as a Service' rights...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq\Logs...
Adding URL reservation at https://seq.xxx.com:port/ for xxx\Administrator...

URL reservation successfully added

[SC] CreateService FAILED 1073:

The specified service already exists.

Could not install the service: Service setup failed

**Reply by nblumhardt** [reply]
Great! That should just mean that you need to uninstall the default instance first:

```
seq service uninstall
seq service install --listen="https://seq.xxx.com:port" -u "xxx\Administrator" -p "xxx"
```

Fingers crossed! 🤞

**Comment by Verbient** [comment]
Still I got the following error:

C:\Windows\system32>seq service uninstall
Uninstalling service...
[SC] DeleteService SUCCESS
Service uninstalled successfully.

C:\Windows\system32>seq service install --listen="https://seq.xxx.com:port" -u "xxx\Administrator" -p "xxx"
Installing service...
Updating the configuration found in C:\ProgramData\Seq\Seq.json...
Ensuring xxx\Administrator is granted 'Log on as a Service' rights...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq\Logs...
Adding URL reservation at https://seq.xxx.com:port/ for xxx\Administrator...

Url reservation add failed, Error: 183
Cannot create a file when that file already exists.

Could not add URL reservation for https://seq.xxx.com:port/: `netsh` returned 1; ignoring
[SC] CreateService SUCCESS
Setting service restart policy...
[SC] ChangeServiceConfig2 SUCCESS
Setting service description...
[SC] ChangeServiceConfig2 SUCCESS
Service installed successfully.

**Reply by nblumhardt** [reply]
Looks like it should be `seq service start` and you're done; we should probably suppress the error messages from `netsh`, the duplicate reservation is fine.

**Reply by Verbient** [reply]
I already tried. Service is not starting:

C:\Windows\system32>seq service start
Starting Seq...
Waiting up to 15 seconds for the service to start (currently: Stopped)...
Started.

C:\Windows\system32>seq service start
Starting Seq...
Waiting up to 15 seconds for the service to start (currently: Stopped)...
Started.

C:\Windows\system32>seq service stop
Cannot stop Seq, current status is: Stopped

**Reply by nblumhardt** [reply]
Thanks for the follow-up. The error causing this should be recorded in C:\ProgramData\Seq\Logs.

**Comment by Verbient** [comment]
Please find a glimpse of the latest log. I masked domain and port for security reasons:

{"@t":"2021-11-01T05:08:06.3161102Z","@mt":"Seq {SeqVersion} running on OS {OSCaption}","SeqVersion":"2021.2.6459","OSCaption":"Microsoft Windows Server 2012 R2 Standard","SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2021-11-01T05:08:06.3640855Z","@mt":"Seq detected {TotalRAM} MB of RAM","TotalRAM":4294.43072,"SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2021-11-01T05:08:06.3854324Z","@mt":"Running in Windows Service host"}
{"@t":"2021-11-01T05:08:06.8760284Z","@mt":"Opening event store at {StreamPath}","StreamPath":"C:\\ProgramData\\Seq\\Stream\\stream.flare","SourceContext":"Flare.Cluster.StorageNodeCoordinator"}
{"@t":"2021-11-01T05:08:06.9109756Z","@mt":"Opening metastore {MetastorePath}","MetastorePath":"C:\\ProgramData\\Seq\\Documents\\default.quince","InstanceName":"Quince","SourceContext":"Quince.Storage.Esent.EsentDocumentStore"}
{"@t":"2021-11-01T05:08:07.1309984Z","@mt":"Storage subsystem available"}
{"@t":"2021-11-01T05:08:07.2531566Z","@mt":"Start","@l":"Error","@x":"Microsoft.AspNetCore.Server.HttpSys.HttpSysException (183): The prefix 'https://seq.xxx.com:Port/' is already registered.\r\n   at Microsoft.AspNetCore.Server.HttpSys.UrlGroup.RegisterPrefix(String uriPrefix, Int32 contextId)\r\n   at Microsoft.AspNetCore.Server.HttpSys.UrlPrefixCollection.RegisterAllPrefixes(UrlGroup urlGroup)\r\n   at Microsoft.AspNetCore.Server.HttpSys.HttpSysListener.Start()","EventId":{"Id":43,"Name":"ListenerStartError"},"SourceContext":"Microsoft.AspNetCore.Server.HttpSys.HttpSysListener"}
{"@t":"2021-11-01T05:08:07.2720832Z","@mt":"Error running the server application","@l":"Fatal","@x":"Microsoft.AspNetCore.Server.HttpSys.HttpSysException (183): The prefix 'https://seq.xxx.com:Port/' is already registered.\r\n   at Microsoft.AspNetCore.Server.HttpSys.UrlGroup.RegisterPrefix(String uriPrefix, Int32 contextId)\r\n   at Microsoft.AspNetCore.Server.HttpSys.UrlPrefixCollection.RegisterAllPrefixes(UrlGroup urlGroup)\r\n   at Microsoft.AspNetCore.Server.HttpSys.HttpSysListener.Start()\r\n   at Microsoft.AspNetCore.Server.HttpSys.MessagePump.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.StartAsync(CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.Start()\r\n   at Seq.Server.ServiceProcess.ServerService.Start()"}
{"@t":"2021-11-01T05:08:07.2746713Z","@mt":"The Seq server could not be started","@l":"Fatal","@x":"Microsoft.AspNetCore.Server.HttpSys.HttpSysException (183): The prefix 'https://seq.xxx.com:Port/' is already registered.\r\n   at Microsoft.AspNetCore.Server.HttpSys.UrlGroup.RegisterPrefix(String uriPrefix, Int32 contextId)\r\n   at Microsoft.AspNetCore.Server.HttpSys.UrlPrefixCollection.RegisterAllPrefixes(UrlGroup urlGroup)\r\n   at Microsoft.AspNetCore.Server.HttpSys.HttpSysListener.Start()\r\n   at Microsoft.AspNetCore.Server.HttpSys.MessagePump.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.StartAsync(CancellationToken cancellationToken)\r\n   at Microsoft.AspNetCore.Hosting.WebHost.Start()\r\n   at Seq.Server.ServiceProcess.ServerService.Start()\r\n   at Seq.Server.ServiceProcess.SeqWindowsService.<>c__DisplayClass2_0.<.ctor>b__0()"}

**Reply by nblumhardt** [reply]
Thanks for the follow-up. Is there something set up in IIS on the machine at the same hostname:port? For Seq to serve from that hostname and port, there should not be any other website on that port in IIS. Does this narrow it down?

**Comment by ryanbuening** [comment]
Just as a data point, we are using Seq on IIS and hosting at seq.mydomain.com. I'm not sure what kind of config was needed as I didn't do that work, but I can try to get the person that did to comment if needed.

**Reply by Verbient** [reply]
@ryanbuening that would be great, thanks. @nblumhardt I have tried once more. I have unloaded/cleared the domain and now there is nothing running in seq.xxxx.com. After that seq started however it was not logging. So I reinstalled the service once more with some modification and since then it is not running any more. The final time I tried with the following parameters without success and here is the outcome.

C:\Windows\system32>seq service install --listen="https://seq.xxxx.com/" -u "xxx\administrator" -p "xxxx"
Installing service...
Updating the configuration found in C:\ProgramData\Seq\Seq.json...
Ensuring xxx\administrator is granted 'Log on as a Service' rights...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq...
Granting Seq.Server.Cli.Features.ServiceCredentialsFeature rights to C:\ProgramData\Seq\Logs...
Adding URL reservation at https://seq.xxxx.com/ for xxx\administrator...

Url reservation add failed, Error: 87
The parameter is incorrect.


Could not add URL reservation for https://seq.xxxx.com/: `netsh` returned 1; ignoring
[SC] CreateService SUCCESS
Setting service restart policy...
[SC] ChangeServiceConfig2 SUCCESS
Setting service description...
[SC] ChangeServiceConfig2 SUCCESS
Service installed successfully.

C:\Windows\system32>seq service start
Starting Seq...
Waiting up to 15 seconds for the service to start (currently: Stopped)...
Started.

Logs as as follows:

{"@t":"2021-11-03T16:12:54.1194909Z","@mt":"Seq {SeqVersion} running on OS {OSCaption}","SeqVersion":"2021.2.6459","OSCaption":"Microsoft Windows Server 2012 R2 Standard","SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2021-11-03T16:12:54.2036777Z","@mt":"Seq detected {TotalRAM} MB of RAM","TotalRAM":4294.43072,"SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
{"@t":"2021-11-03T16:12:54.2356261Z","@mt":"Running in Windows Service host"}
{"@t":"2021-11-03T16:12:54.8418369Z","@mt":"Opening event store at {StreamPath}","StreamPath":"C:\\ProgramData\\Seq\\Stream\\stream.flare","SourceContext":"Flare.Cluster.StorageNodeCoordinator"}
{"@t":"2021-11-03T16:12:54.8853107Z","@mt":"Opening metastore {MetastorePath}","MetastorePath":"C:\\ProgramData\\Seq\\Documents\\default.quince","InstanceName":"Quince","SourceContext":"Quince.Storage.Esent.EsentDocumentStore"}
{"@t":"2021-11-03T16:12:55.0655426Z","@mt":"Storage subsystem available"}
{"@t":"2021-11-03T16:12:55.2176518Z","@mt":"Seq listening on {ListenUris}","ListenUris":["https://seq.xxxx.com/"]}
{"@t":"2021-11-03T16:13:55.3585554Z","@mt":"{CollectionCount} more generation 2 garbage collection(s) occurred","CollectionCount":1,"SourceContext":"Seq.Server.Features.Diagnostics.GCLogger"}
{"@t":"2021-11-03T16:17:55.3300234Z","@mt":"Metrics sampled","CacheDays":0,"CacheEvents":0,"SystemMemoryUtilization":0.6479166766020154,"ProcessWorkingSetBytes":92909568,"ProcessThreads":27,"ArrivedEventsPerMinute":0,"IngestedEventsPerMinute":0,"IngestedBytesPerMinute":0,"InvalidPayloadsPerMinute":0,"UnauthorizedPayloadsPerMinute":0,"BroadcastQueueCount":0,"BroadcastSubscriptionQueuesAggregateCount":0}
{"@t":"2021-11-03T16:22:55.3135381Z","@mt":"Metrics sampled","CacheDays":0,"CacheEvents":0,"SystemMemoryUtilization":0.5982812641578719,"ProcessWorkingSetBytes":95010816,"ProcessThreads":29,"ArrivedEventsPerMinute":0,"IngestedEventsPerMinute":0,"IngestedBytesPerMinute":0,"InvalidPayloadsPerMinute":0,"UnauthorizedPayloadsPerMinute":0,"BroadcastQueueCount":0,"BroadcastSubscriptionQueuesAggregateCount":0}
{"@t":"2021-11-03T16:22:55.3242966Z","@mt":"Applying {Count} retention policies","Count":0,"SourceContext":"Seq.Server.Features.Retention.RetentionPolicyApplier"}
{"@t":"2021-11-03T16:22:55.3310766Z","@mt":"Retention processing and compaction took {Elapsed} ms; allocating {Allocation} ms for indexing","Elapsed":12.0786,"Allocation":599987.9214}
{"@t":"2021-11-03T16:22:55.3340204Z","@mt":"Beginning indexing","SourceContext":"Seq.Server.Features.Indexing.EventStoreIndexer"}
{"@t":"2021-11-03T16:22:55.6053239Z","@mt":"Wrote {IndexSetCount} index sets in {Elapsed:0.000} ms","@r":["270.000"],"IndexSetCount":0,"Elapsed":270,"SourceContext":"Seq.Server.Features.Indexing.EventStoreIndexer"}
{"@t":"2021-11-03T16:22:55.6070044Z","@mt":"Indexing took {Elapsed} ms; total iteration time was {Iteration} ms","Elapsed":275,"Iteration":287.6226}
{"@t":"2021-11-03T16:27:55.3285696Z","@mt":"Metrics sampled","CacheDays":0,"CacheEvents":0,"SystemMemoryUtilization":0.5988182498843526,"ProcessWorkingSetBytes":97058816,"ProcessThreads":27,"ArrivedEventsPerMinute":0,"IngestedEventsPerMinute":0,"IngestedBytesPerMinute":0,"InvalidPayloadsPerMinute":0,"UnauthorizedPayloadsPerMinute":0,"BroadcastQueueCount":0,"BroadcastSubscriptionQueuesAggregateCount":0}
{"@t":"2021-11-03T16:32:55.3138988Z","@mt":"Applying {Count} retention policies","Count":0,"SourceContext":"Seq.Server.Features.Retention.RetentionPolicyApplier"}

**Reply by nblumhardt** [reply]
Thanks for the update. Have you bound an SSL certificate for your hostname/port combination using `seq bind-ssl`?

**Reply by Verbient** [reply]
I have bound SSL with the parent domain www.xxx.com with wildcard support, instead of seq.xxx.com. However SSL is not the issue right now. I tried registering **http**.seq.xxx.com, but that too didn't work. Probably is might have something to do with **duplicate reservation** as mentioned by you in [earlier in this discussion](https://github.com/datalust/seq-tickets/discussions/1360#discussioncomment-1567816) . There should be a way seq cli should flag more details about the duplicate reservation on screen. Desperately looking forward to your help. Thanks.

**Reply by Verbient** [reply]
Bingo!!! it started and it's logging. I don't know how. But I tried adding a subdomain in PLESK seq.xxx.com. But it returned an error saying "...Another website may be using the same port..." After this SEQ  started running fine automatically. 

@nblumhardt thanks for your persistent help. May I suggest that you should update the SEQ documentation on the page https://docs.datalust.co/docs/urls and make it little bit more comprehensive. You could update the seq server install command to emphasize the use of "COMPUTERNAME\username" and also mention that the domain seq.xxx.com should not be preoccupied. People may think that they should create an empty domain and SEQ would start writing on it. Also you may put some CLI error code/message description so that end user can themselves interpret the error with seeking help anywhere.

I would now try to configure SSL and add a custom username and password to the seq page seq.xxx.com.

**Reply by nblumhardt** [reply]
🎉 🎉 🎉 🎉 🎉 

Thanks for the feedback, will work on those things 👍

**Reply by Verbient** [reply]
I am trying to bind SSL, but getting the following error:

C:\Windows\system32>seq bind-ssl --thumbprint="?xxx"
Binding the Seq server on *:443 to SSL certificate 20f31dxxx6d

SSL Certificate add failed, Error: 1312
A specified logon session does not exist. It may already have been terminated.

Failed to bind the certificate: `netsh` returned exit code 1.
